### PR TITLE
syncval: Set queue submit time sync val to default on

### LINKED ISSUE #7127: syncval: Set VALIDATION_CHECK_ENABLE_SYNCHRONIZATION_VALIDATION_QUEUE_SUBMIT to be on by default
Now that this feature has been implemented for several months and SDK releases, it seems like it should be on by default.

### PR DESCRIPTION
**Pull request by jzulauf-lunarg**
Set queue submit time sync validation to be on by default.

Add disable setting for those that want to turn it off for performance.

Fixes #7127

### CODE REVIEW

**Review by spencer-lunarg — APPROVED** [review]

File: layers/layer_options.cpp [hunk 1]
```diff
@@ -137,9 +140,6 @@ void SetValidationEnable(CHECK_ENABLED &enable_data, const ValidationCheckEnable
             enable_data[vendor_specific_img] = true;
             enable_data[vendor_specific_nvidia] = true;
             break;
-        case VALIDATION_CHECK_ENABLE_SYNCHRONIZATION_VALIDATION_QUEUE_SUBMIT:
-            enable_data[sync_validation_queue_submit] = true;
-            break;
```
Comment (christophe-lunarg) [inline layers/layer_options.cpp:142]: It's not clear to me why stuff move around but that's ok with me. Just commenting to be sure that's not an accident,

Comment (christophe-lunarg) [reply layers/layer_options.cpp:142]: Oh, it went from an enable to a disable ok!

File: tests/unit/sync_val_positive.cpp [hunk 2]
```diff
@@ -27,18 +27,8 @@ void VkSyncValTest::InitSyncValFramework(bool enable_queue_submit_validation) {
...
-        {OBJECT_LAYER_NAME, "enables", VK_LAYER_SETTING_TYPE_STRING_EXT, 1, kEnableQueuSubmitSyncValidation}};
-    // The pNext of qs_settings is modified by InitFramework that's why it can't
-    // be static (should be separate instance per stack frame). Also we show
-    // explicitly that it's not const (InitFramework casts const pNext to non-const).
-    VkLayerSettingsCreateInfoEXT qs_settings{VK_STRUCTURE_TYPE_LAYER_SETTINGS_CREATE_INFO_EXT, nullptr,
-                                             static_cast<uint32_t>(std::size(settings)), settings};
-    if (enable_queue_submit_validation) {
-        features_.pNext = &qs_settings;
-    }
+    // QueueSubmit Validation *always* on (default behavior)
```
Comment (christophe-lunarg) [inline tests/unit/sync_val_positive.cpp:30]: It sounds like to me we need to a test to check the "disable" code path is working? If it's always on, it's never tested,

**Review by artem-lunarg — APPROVED** [review]

File: tests/unit/sync_val_positive.cpp (hunk 2 above)
[the remark this answers is not in the document]
Comment (jzulauf-lunarg) [reply tests/unit/sync_val_positive.cpp]: Cleaned this up...

Comment (jzulauf-lunarg) [reply tests/unit/sync_val_positive.cpp]: done

File: layers/layer_options.cpp [hunk 3]
```diff
@@ -477,6 +475,8 @@ void ProcessConfigAndEnvSettings(ConfigAndEnvSettings *settings_data) {
         SetValidationSetting(layer_setting_set, settings_data->disables, object_tracking, SETTING_OBJECT_LIFETIME);
         SetValidationSetting(layer_setting_set, settings_data->disables, shader_validation, SETTING_CHECK_SHADERS);
         SetValidationSetting(layer_setting_set, settings_data->disables, shader_validation_caching, SETTING_CHECK_SHADERS_CACHING);
+        SetValidationSetting(layer_setting_set, settings_data->disables, sync_validation_queue_submit,
```
Comment (christophe-lunarg) [inline layers/layer_options.cpp:478]: This is the part of the new settings and it looks correct.

**Review by christophe-lunarg — CHANGES_REQUESTED** [review]
Sorry, I just realized : You only modified the legacy settings, that could be configured with VK_EXT_validation_features and the flags, but there is a new approach:
```
                            "settings": [
                                {
                                    "key": "sync_queue_submit",
                                    "label": "QueueSubmit Synchronization Validation",
                                    "description": "Enable synchronization validation between submitted command buffers when Synchronization Validation is enabled. This option will increase the synchronization performance cost.",
                                    "type": "BOOL",
                                    "default": false,
                                    "status": "ALPHA",
                                    "dependence": {
                                        "mode": "ALL",
                                        "settings": [
                                            {
                                                "key": "validate_sync",
                                                "value": true
                                            }
                                        ]
                                    }
                                }
                            ]
                            ```
        Replace this by :
                 ```
                            "settings": [
                                {
                                    "key": "sync_queue_submit",
                                    "label": "QueueSubmit Synchronization Validation",
                                    "description": "Enable synchronization validation between submitted command buffers when Synchronization Validation is enabled. This option will increase the synchronization performance cost.",
                                    "type": "BOOL",
                                    "default": true,
                                    "status": "STABLE",
                                    "dependence": {
                                        "mode": "ALL",
                                        "settings": [
                                            {
                                                "key": "validate_sync",
                                                "value": true
                                            }
                                        ]
                                    }
                                }
                            ]
                            ```           
                            
                            Sorry for the late feedback :/

**Review by christophe-lunarg — APPROVED** [review]

**Pull request merged**